### ISSUE TITLE
series chart does not work as range chart

### ISSUE THREAD

**Issue by hudannis** [dc-js/dc.js#479]
Hi, I am using seriesChart to draw multiple lines in one chart with a selecting chart. But it seems seriesChart doesn't work with rangeChart. I select a range by brush but the main chart doesn't change as expected.
expected image:
![expected](https://f.cloud.github.com/assets/5782042/1740836/8a93793c-63d4-11e3-8682-0a1c00dedf0f.png)

get the following image:
![notchange](https://f.cloud.github.com/assets/5782042/1740839/ae6081c0-63d4-11e3-9182-c24e29204470.png)

Moreover, if the mainchart can show the selected range of the indicator chart, can the data be selected correctly? Because the dimension for the seriesChart has two keys, while the filter getting by the selected range has only one key.

**Comment by peter-houck** [comment]
I tried hacking series.html example to add a rangeChart.  The brush still does not filter the main chart.  Any ideas? Here is the hacked up code:

`<!DOCTYPE html>
<html lang="en">
<head>
    <title>dc.js - Series Example</title>
    <meta charset="UTF-8">
    <link rel="stylesheet" type="text/css" href="../css/bootstrap.min.css">
    <link rel="stylesheet" type="text/css" href="../css/dc.css"/>
</head>
<body>

<div class="container">
<script type="text/javascript" src="header.js"></script>
<div id="test"></div>
<div id="test-overview"></div>
<button onclick="button1()">1</button>
<button onclick="button2()">2</button>
<button onclick="button3()">3</button>

<script type="text/javascript" src="../js/d3.js"></script>
<script type="text/javascript" src="../js/crossfilter.js"></script>
<script type="text/javascript" src="../js/dc.js"></script>
<script type="text/javascript">

var chart = dc.seriesChart("#test");
var overviewChart = dc.seriesChart("#test-overview");
var ndx, runDimension, runGroup, overviewRunDimension, overviewRunGroup;

d3.csv("morley.csv", function(error, experiments) {

  ndx = crossfilter(experiments);
  runDimension = ndx.dimension(function(d) {return [+d.Expt, +d.Run]; });
  overviewRunDimension = ndx.dimension(function(d) {return [+d.Expt, +d.Run]; });
  runGroup = runDimension.group().reduceSum(function(d) { return +d.Speed; });
  overviewRunGroup = overviewRunDimension.group().reduceSum(function(d) { return +d.Speed; });

  chart
    .width(768)
    .height(480)
    .chart(function(c) { return dc.lineChart(c).interpolate('basis'); })
    .x(d3.scale.linear().domain([0,20]))
    .brushOn(false)
    .yAxisLabel("Measured Speed km/s")
    .xAxisLabel("Run")
    .clipPadding(10)
    .elasticY(true)
    .dimension(runDimension)
    .group(runGroup)
    .mouseZoomable(true)
    .rangeChart(overviewChart)
    .seriesAccessor(function(d) {return "Expt: " + d.key[0];})
    .keyAccessor(function(d) {return +d.key[1];})
    .valueAccessor(function(d) {return +d.value - 500;})
    .legend(dc.legend().x(350).y(350).itemHeight(13).gap(5).horizontal(1).legendWidth(140).itemWidth(70));
  chart.yAxis().tickFormat(function(d) {return d3.format(',d')(d+299500);});
  chart.margins().left += 40;

  overviewChart
    .width(768)
    .height(100)
    .chart(function(c) { return dc.lineChart(c).interpolate('basis'); })
    .x(d3.scale.linear().domain([0,20]))
    .brushOn(true)
    .xAxisLabel("Run")
    .clipPadding(10)
    .dimension(overviewRunDimension)
    .group(overviewRunGroup)
    .seriesAccessor(function(d) {return "Expt: " + d.key[0];})
    .keyAccessor(function(d) {return +d.key[1];})
    .valueAccessor(function(d) {return +d.value - 500;})

  dc.renderAll();

});

function load_button(file) {
    return function load_it() {
        d3.csv(file, function(error, experiments) {
            ndx.remove();
            ndx.add(experiments);
            dc.redrawAll();
        });
    };
}

var button1 = load_button("morley.csv"),
    button2 = load_button("morley2.csv"),
    button3 = load_button("morley3.csv");

</script>

</div>
</body>
</html>`

**Comment by peter-houck** [comment]
I hacked series.html to have a rangeChart.  But I still cannot get the brush filter to work.  Any ideas?  Here is the salient code:

`ndx = crossfilter(experiments)`

**Comment by peterh2** [comment]
Here is a jsfiddle link:
[range and series charts](http://jsfiddle.net/peterth/j3koz28f/2/)

**Comment by gordonwoodhull** [comment]
Hi @peterh2, you're right, this issue should not have been closed just because the original asker disappeared.

**Comment by gordonwoodhull** [comment]
I have been meaning to look at your SO question and to see if there is something that needs to be changed here in dc.js. Not sure if I will have time today. Thanks for posting a fiddle. I expect that what is going on is that the dimensions are incompatible and a [filterHandler](https://github.com/dc-js/dc.js/blob/develop/web/docs/api-latest.md#dc.baseMixin+filterHandler) or something needs to be applied to translate the coordinates.

**Comment by gordonwoodhull** [comment]
This is really an instance of #390, just slightly less complicated.

**Comment by peterh2** [comment]
Thanks for looking in to this.  Do you think that you will have some sort of  fix for this any time soon?  I would really like to use the 2.0 series chart if possible rather than reverting back to  the 1.75 composite charts.  If you think it is fixable in the next month or so, I will continue with 2.0.

**Comment by gordonwoodhull** [comment]
Unfortunately I am deep underwater in technical debt and can't make a promise about when this will be fixed. 

In response to your SO question I came up with a pretty complex workaround, but it exposes other bugs in dc.js - in particular #949 points just off the edge get clipped, which makes this example look particularly hideous. 

I'll publish my answer soon, still needs some explanation. Don't feel obligated to accept the answer as it is quite messy!

**Comment by gordonwoodhull** [comment]
#949 will definitely be fixed for 2.0 and I'm hoping to get to that in the next week or two. 

So, there is a solution, it's not a great one.

The solution for this issue right here is quite a bit more complex and I don't have any ETA for a solution.

**Comment by peterh2** [comment]
Cool.  For now I will try to work with the SO / jsFiddle "solution" and just wait for [#949]https://github.com/dc-js/dc.js/issues/949) to get fixed.  Definitely appreciate your work on this!

**Comment by gordonwoodhull** [comment]
I added an example based on the workaround I explained in this SO answer to @peterh2's question on the same topic, [dc.js seriesChart doesn't seem to work with the rangeChart](http://stackoverflow.com/a/38058859/676195):

http://dc-js.github.io/dc.js/examples/range-series.html

It's pretty ugly but conceptually it's sound. Basically it makes one of the child charts the range chart instead of the whole composite chart. Hmmmm, maybe that would have been an easier solution.

The trickier problem was that the composite chart was filtering each of its children *twice*, doing the filter null + filter range thing. Sometimes it seemed to land in the wrong place and end up with the filter null when there was a brush. This was inefficient anyway so I replaced it with a replaceFilter in the above commit.
